common/map.h

```diff
@@ -365,6 +373,13 @@ class MapBase : protected RawHashtable::BaseImpl<InputKeyT, InputValueT,
 // allocations the performance of hashtable routines may be unacceptably bad and
 // another data structure or key design is likely preferable.
 //
+// Note that `Map` has "shallow" const semantics: a `const Map<K, V>&` can't be
```

Comment:
Maybe just say both `Map` and `MapBase` here to avoid duplicating the content?

Also, would it make sense to more explain these by analogy "much like `MapView` ...", and nudge users towards using `MapView` rather than a const-qualified `Map` or `MapBase` reference when reasonable?